x-pack/plugins/security_solution/common/endpoint/models/event.ts

```diff
@@ -52,3 +52,136 @@ export function parentEntityId(event: ResolverEvent): string | undefined {
   }
   return event.process.parent?.entity_id;
 }
+
+/**
+ * @param event The event to get the category for
+ */
+export function primaryEventCategory(event: ResolverEvent): string {
+  // Returning "Process" as a catch-all here because it seems pretty general
+  let eventCategoryToReturn: string = 'Process';
+  if (isLegacyEvent(event)) {
+    const legacyFullType = event.endgame.event_type_full;
+    if (legacyFullType) {
+      return legacyFullType;
+    }
+  } else {
+    const eventCategories = event.event.category;
+    const category =
+      typeof eventCategories === 'string' ? eventCategories : eventCategories[0] || '';
+
+    if (category) {
+      eventCategoryToReturn = category;
+    }
+  }
+  return eventCategoryToReturn;
+}
+
+/**
+ * ECS event type will be things like 'creation', 'deletion', 'access', etc.
+ * see: https://www.elastic.co/guide/en/ecs/current/ecs-event.html
+ * @param event The ResolverEvent to get the ecs type for
+ */
+export function ecsEventType(event: ResolverEvent): string[] {
+  if (isLegacyEvent(event)) {
+    return [event.endgame.event_subtype_full || ''];
+  }
+  return typeof event.event.type === 'string' ? [event.event.type] : event.event.type;
+}
+
+/**
+ * #Descriptive Names For Related Events:
+ *
+ * The following section provides facilities for deriving **Descriptive Names** for ECS-compliant event data.
+ * There are drawbacks to trying to do this: It *will* require ongoing maintenance. It presents temptations to overarticulate.
+ * On balance, however, it seems that the benefit of giving the user some form of information they can recognize & scan outweighs the drawbacks.
+ */
+
+/**
+ * A type that annotates the basic requirements for giving the event a `descriptive name`
+ */
+type detailRecord<T> = T extends 'registry'
```

Comment:
Per our discussion, let's replace this w/ https://github.com/elastic/kibana/commit/c14226e697ca1309c5abad4650e39691a2335053